.github/workflows/pull_request.yml

```diff
@@ -47,36 +47,26 @@ jobs:
             ${{ runner.os }}-pact-
             ${{ runner.os }}-
 
-      # Manually running install tools instead of scripts/install_deps.sh
-      # Tools referenced in install_deps.sh script require Xcode 12.2 on macOS 10.15.
       - name: Prepare the tools
         run: |
-          brew update && brew bundle
-          carthage checkout
+          scripts/install_deps.sh
 
-      # Manually running xcodebuild instead of scripts/build.sh
-      # Tools referenced in install_deps.sh script require Xcode 12.2 on macOS 10.15.
       - name: "Run tests"
         run: |
-          ./scripts/carthage_xcode12 update --platform $CARTHAGE_PLATFORM
-          set -o pipefail && xcodebuild clean test -project $PROJECT_NAME -scheme "$SCHEME" -destination "$DESTINATION" -configuration Debug GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=YES GCC_GENERATE_TEST_COVERAGE_FILES=YES
-
-      - name: "Upload code coverage"
-        run: |
-          bash <(curl -s https://codecov.io/bash) -J 'PactConsumerSwift'
+          scripts/build.sh
 
   test_spm:
-    name: Test for SPM compatibility
-    runs-on: macOS-10.15
+    name: Test for SPM
+    runs-on: macOS-latest
 
     steps:
       - name: Checkout repository
         uses: actions/checkout@v2
         with:
           submodules: recursive
 
-      - name: Use Xcode 11.6
-        run: sudo xcode-select -switch /Applications/Xcode_11.6.app
+      - name: Use Xcode 12.2
+        run: sudo xcode-select -switch /Applications/Xcode_12.2.app
```

Comment:
If we are using macOS-latest but hardcoding Xcode versions won't it get out of synch? i.e. in future 11.X is released which doesn't have 12.2 of XCode installed.

Reply:
As it stands now, `macOS-11` runners have the following Xcode versions available:
- 11.7
- 12.2 (default)
- 12.3

[https://github.com/actions/virtual-environments/blob/main/images/macos/macos-11.0-Readme.md](https://github.com/actions/virtual-environments/blob/main/images/macos/macos-11.0-Readme.md)

Current `macOS-default` runner have twelve Xcode versions installed on them of which 12.2 is still the default one. Based on this wide support for multiple versions I would assume `macOS-11` will continue to support 12.2 version in the near future. When `macOS-11` becomes the `default` and if 12.2 is not available anymore on these runners anymore, it's fairly easy to change the version.

Reply:
Got me thinking and I agree that going forward we can explicitly specify which OS version to run on and rely on the "latest" defaults to run on previous versions' agents. This will reduce the need for us to catch up and fix things down the line.